### PR TITLE
Feature/v1-193 | Get employee's available courses API.

### PR DESCRIPTION
**Pull request by dapodev**
# Overview
New endpoint implemented. Some utils added, some refactored.
# Changes
- Backend
1. New API endpoint `[GET] /api/employees/:id/courses/available?title=jq` where `:id` is employee's id and `title` is query parameter to filter courses by their title if necessary. *NO PAGINATION*
2. Compare `ObjectId` util refactored so to allow compare ids with passing one or both `string` values.
3. Implemented available courses filter util and normalizer.
4. Implemented new provider to get courses with statuses without pagination.
- Frontend
1. Refactored search logic at AddCourseDialog for Employee at Employees page.
2. Added api at frontend for getting available courses
3. Refactored types

### CODE REVIEW

**Review by irina-sinitsyna — DISMISSED** [review]

**Review by irina-sinitsyna — APPROVED** [review]

File: frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CourseItem/CourseItem.tsx [hunk 1]
```diff
@@ -10,17 +10,10 @@ interface IProps {
...
 
-const CourseItem: FC<IProps> = ({
-  checkboxValue,
-  isChecked,
-  title,
-  handleCheckboxChange,
-  courseRef,
-}) => (
-  <StyledItem ref={courseRef}>
+const CourseItem: FC<IProps> = ({ checkboxValue, isChecked, title, handleCheckboxChange }) => (
```
Comment (PechurinaLena) [inline frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CourseItem/CourseItem.tsx:15]: Could we change interface name as ICourseItemProps?

File: frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CoursesList.tsx [hunk 2]
```diff
@@ -1,20 +1,19 @@
 import { FC, BaseSyntheticEvent } from 'react';
 
-import { ICourse, CoursesListType, TCheckedCourse } from 'types/course';
+import { LOADER } from 'constants/loaderTypes';
+import { IShortCourseInfo, CoursesListType } from 'types/course';
 import Loader from 'components/Loader';
-import { Loaders } from 'enums/loader';
 
 import { CourseItem } from './CourseItem';
 import { StyledList, StyledHorizontalLine } from './styled';
```
Comment (PechurinaLena) [inline frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CoursesList.tsx:9]: and interface name here

File: frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CourseItem/CourseItem.tsx (hunk 1 above)
Comment (ostapdans99) [reply frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CourseItem/CourseItem.tsx:15]: We dont have one solution for this cases. Somebody wants to create props interfaces inside types.ts, somebody inside component file. In this case i just removed on incoming prop, Before changing this case, we should found one solution. We can discuss it on ower meeting in Tuesday

File: frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CoursesList.tsx (hunk 2 above)
Comment (ostapdans99) [reply frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CoursesList.tsx:9]: I dont understand what you meen

Comment (PechurinaLena) [reply frontend/src/pages/EmployeeProfile/EmployeeSkillsAndCourses/AddCourseButton/AddCourseDialog/CoursesList/CoursesList.tsx:9]: wrong displayed here, i leave a comment on a IProps interface name, but I got you decision. lets discuss about interface naming on a next meeting

**Review by PechurinaLena — APPROVED** [review]

**Pull request merged**